src/Rules/BinaryDist.hs

```diff
@@ -61,6 +63,27 @@ bindistRules = do
     -- generate the Makefile that enables the "make install" part
     root -/- "bindist" -/- "ghc-*" -/- "Makefile" %> \makefilePath ->
       writeFile' makefilePath bindistMakefile
+    
+    root -/- "bindist" -/- "ghc-*" -/- "wrappers/ghc" %> \wrapperGhcPath -> 
+      writeFile' wrapperGhcPath ghcWrapper
+
+    root -/- "bindist" -/- "ghc-*" -/- "wrappers/ghc-pkg" %> \wrapperGhcPkgPath ->
+      writeFile' wrapperGhcPkgPath ghcPkgWrapper
+
+    root -/- "bindist" -/- "ghc-*" -/- "wrappers/ghci" %> \wrapperGhciPath ->
+      writeFile' wrapperGhciPath ghciWrapper
+
+    root -/- "bindist" -/- "ghc-*" -/- "wrappers/haddock" %> \wrapperHaddockPath ->
+      writeFile' wrapperHaddockPath haddockWrapper
+
+    root -/- "bindist" -/- "ghc-*" -/- "wrappers/common" %> \wrapperCommonPath ->
+      writeFile' wrapperCommonPath commonWrapper
+
+    root -/- "bindist" -/- "ghc-*" -/- "wrappers/hsc2hs" %> \wrapperHsc2hsPath ->
+      writeFile' wrapperHsc2hsPath hsc2hsWrapper
+
+    root -/- "bindist" -/- "ghc-*" -/- "wrappers/runhaskell" %> \wrapperRunHaskellPath ->
+      writeFile' wrapperRunHaskellPath runhaskellWrapper
```

Comment:
I think you could substantially shorten this bit of code by having an function to abstract over the general pattern:

``` haskell
  bindistWrapperRule root "ghc" wrapperGhcPath ghcWrapper
  [...]
  bindistWrapperRule root "hsc2hs" wrapperHsc2hsPath hsc2hsWrapper
  bindistWrapperRule root "runhaskell" wrapperRunHaskellPath runhaskellWrapper
```